components/styles/styler.component.ts

```diff
@@ -402,17 +417,13 @@ export class HsStylerComponent {
     this.hasPoint = src.hasPoint;
     this.layerTitle = this.HsStylerService.layer.get('title');
   }
-
   readCurrentStyle(layer: VectorLayer): void {
-    let style = layer.getStyle();
-    if (this.HsLayerUtilsService.isLayerClustered(layer)) {
-      style = layer.get('hsOriginalStyle');
-    }
-    if (typeof style == 'function') {
-      const resolvedStyle: Style | Style[] = style(new Feature());
+    const resolvedStyle = this.HsStylerService.getLayerStyleObject(layer);
```

Comment:
Will this work even if we dont have features currently in the layer or we don't we are not interested in that since hasLine/poly/point will have false anyways?

Reply:
If should work, because the getLayerStyleObject, gets layer style if exists, and if style is function if creates an empty new feature, for it to work.